src/statement.d

```diff
@@ -5149,6 +5154,19 @@ public:
                 }
             }
         }
+
+        if (sc.func)
+        {
+            flags |= sc.func.flags & (FUNCFLAGcppExceptions | FUNCFLAGdExceptions);
+            sc.func.flags |= flags;
+            if (flags == (FUNCFLAGcppExceptions | FUNCFLAGdExceptions))
+            {
+                error("cannot mix catching D and C++ exceptions in the same function");
```

Comment:
> You can e.g. just look at the language code for the exception to distinguish between them.

Of course, but that's only half the problem. The other half is the pointer in the TType table - a TypeInfo pointer needs to be distinguished from a C++ RTTI pointer. Elie suggests wrapping the C++ info in a TypeInfo. That may work for dmd, I have not investigated it completely yet. If it doesn't work, or is impractical, the other way to distinguish what kind of pointers are in TType is to use a different personality function that sets a flag.